### ISSUE TITLE
[Accessibility] Dialog cancel button missing aria-label

### ISSUE THREAD

**Issue by kesane-msft** [OfficeDev/office-ui-fabric-react#1863]
### Bug Report

- __Package version(s)__:  Latest
- __Browser and OS versions__: IE Edge, Windows 10 

#### Priorities and help requested (not applicable if asking question):

Are you willing to submit a PR to fix? No

Requested priority: High

Products/sites affected: (if applicable)

#### Describe the issue:
The name of the cancel button at top of dialog in the "Inspect" tool is missing. 

#### Actual behavior:
Open the dialog on the Office fabric components in Edge browser.
Have the [Inspect tool](https://msdn.microsoft.com/en-us/library/windows/desktop/dd318521(v=vs.85).aspx)  open.
Focus on the Cancel button at top of dialog. The name field is empty in the tool.

#### Expected behavior:

Name field should be populated as Cancel.

### If applicable, please provide a codepen repro:

![accessibility - cancel button name](https://cloud.githubusercontent.com/assets/28951219/26449579/9f5cd930-4170-11e7-913e-3490399f40f9.png)

**Comment by micahgodbolt** [comment]
Can you point to some accessibility documentation that recommends using the `name` value in this way?

Most discussion I see around `name` is that it is used on form submission.

https://stackoverflow.com/questions/1397592/difference-between-id-and-name-attributes-in-html

**Comment by mdahamiwal** [comment]
@micahgodbolt, the name property is UIA property, it maps to text set using aria-label, aria-labelledby in HTML. The `name` we use in HTML is different. Here if name property is empty the screen reader will not announce it and it will only say button. We should provide it as `close` for dialogs.

**Comment by micahgodbolt** [comment]
@mdahamiwal so the bug here is a missing aria-label, right? That I'm happy to add.

**Comment by mdahamiwal** [comment]
Yup! Thanks Micah :)

**Comment by micahgodbolt** [comment]
Ahh, looks like we already have `closeButtonAriaLabel`  which wasn't being set in our demo page. We obviously want to make sure the string can be internationalized.

I'll update the example, and it'll go out with the next site push.

**Comment by micahgodbolt** [comment]
heh...ok, it seems SOME code was there supporting closeButtonAriaLabel, but after some recent refactoring, it wasn't working properly. Working on it now.

**Comment by mdahamiwal** [comment]
@micahgodbolt, just for the record, what is the frequency we update the official site? Also, what about https://fabricreact.azurewebsites.net/fabric-react/master/ ? As name suggest, It would be great if we can update it regularly to match our master branch payload. This will enable our customers to see the current state of components as it is and avoid filling of duplicate issues that are already fixed in master and closed.

**Comment by nirmalapemireddy** [comment]
@micahgodbolt  : When are you going to merge this fix ?

**Comment by micahgodbolt** [comment]
@nirmalapemireddy Hopefully really soon! Was just waiting for some final feedback on how I approached the larger change that I did.
